### ISSUE TITLE
Spring Integration 5.x and Spring Boot 2.x support

### ISSUE THREAD

**Issue by prafsoni** [spring-projects/spring-integration-extensions#185]
Any roadmap when these extensions will be updated to support spring integration 5.x? I am specifically interested in "spring-integration-smb"

**Comment by artembilan** [comment]
We do that on demand. As far as no any feature plans in these project we just support them in respect to a general community request.

Each of the extension is a separate project, so there is no need to wait for all of them to be upgraded.

Since you have request this one, we will consider to address it this week.

The contribution is welcome though! And you are free to raise an upgrade PR for the SMB module and that will expedite its release. :smile:

**Comment by prafsoni** [comment]
@artembilan I will give it a try tonight.

**Comment by prafsoni** [comment]
@artembilan hi I made first attempt to upgrade the project to spring integration 5, and ran into issues with `SmbInboundChannelAdapterParser ` due to addition of ` protected abstract Class<? extends AbstractPersistentAcceptOnceFileListFilter<?>> getPersistentAcceptOnceFileListFilterClass()` in `AbstractRemoteFileInboundChannelAdapterParser `

**Comment by artembilan** [comment]
Right, we have introduce since there a fine-grained filters configuration based on the target remote file info.

You can take a look into the Spring Integration FTP and implement similar filters for the SMB: https://github.com/spring-projects/spring-integration/blob/master/spring-integration-ftp/src/main/java/org/springframework/integration/ftp/config/FtpInboundChannelAdapterParser.java
